File: src/main/java/de/tectoast/emolga/bot/SlashListener.java
package de.tectoast.emolga.bot;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.GuildCommandEvent;
import net.dv8tion.jda.api.JDA;
import net.dv8tion.jda.api.entities.TextChannel;
import net.dv8tion.jda.api.entities.User;
import net.dv8tion.jda.api.events.ReadyEvent;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import org.jetbrains.annotations.NotNull;

import static de.tectoast.emolga.commands.Command.buildEnumeration;
import static de.tectoast.emolga.commands.Command.sendToMe;
import static de.tectoast.emolga.utils.Constants.FLOID;


public class SlashListener extends ListenerAdapter {
    final JDA jda;

    public SlashListener(JDA jda) {
        this.jda = jda;
    }

    @Override
    public void onReady(@NotNull ReadyEvent e) {

    }

    @Override
    public void onSlashCommand(@NotNull SlashCommandEvent e) {
        Command command = Command.byName(e.getName());
        User u = e.getUser();
        TextChannel tco = e.getTextChannel();
        try {
            new GuildCommandEvent(command, e);
        } catch (Command.MissingArgumentException ex) {
            Command.ArgumentManagerTemplate.Argument arg = ex.getArgument();
            if (arg.hasCustomErrorMessage()) e.reply(arg.getCustomErrorMessage()).queue();
            else {
                e.reply("Das benötigte Argument `" + arg.getName() + "`, was eigentlich " + buildEnumeration(arg.getType().getName()) + " sein müsste, ist nicht vorhanden!\n" +
                        "Nähere Informationen über die richtige Syntax für den Command erhältst du unter `e!help " + command.getName() + "`.").queue();
            }
            if (u.getIdLong() != FLOID) {
                sendToMe("MissingArgument " + tco.getAsMention() + " Server: " + tco.getGuild().getName());
            }
        } catch (Exception ex) {
            ex.printStackTrace();
            e.getHook().sendMessage("Es ist ein Fehler beim Ausführen des Commands aufgetreten!\nWenn du denkst, dass dies ein interner Fehler beim Bot ist, melde dich bitte bei Flo/TecToast.\n" + command.getHelp(e.getGuild()) + (u.getIdLong() == FLOID ? "\nJa Flo, du sollst dich auch bei ihm melden du Kek! :^)" : "")).queue();
        }
    }
}

File: src/main/kotlin/de/tectoast/emolga/bot/SlashListener.kt
package de.tectoast.emolga.bot

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.Command.MissingArgumentException
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.utils.Constants.FLOID
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter

class SlashListener : ListenerAdapter() {
    override fun onSlashCommandInteraction(e: SlashCommandInteractionEvent) {
        val command = Command.byName(e.name)!!
        val u = e.user
        val tco = e.textChannel
        try {
            GuildCommandEvent(command, e)
        } catch (ex: MissingArgumentException) {
            val arg = ex.argument!!
            e.reply(
                arg.customErrorMessage
                    ?: """Das benötigte Argument `${arg.name}`, was eigentlich ${Command.buildEnumeration(arg.type.getName())} sein müsste, ist nicht vorhanden!
Nähere Informationen über die richtige Syntax für den Command erhältst du unter `e!help ${command.name}`.""".trimIndent()
            )
            if (u.idLong != FLOID) {
                Command.sendToMe("MissingArgument " + tco.asMention + " Server: " + tco.guild.name)
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
            e.hook.sendMessage(
                """
    Es ist ein Fehler beim Ausführen des Commands aufgetreten!
    Wenn du denkst, dass dies ein interner Fehler beim Bot ist, melde dich bitte bei Flo/TecToast.
    ${command.getHelp(e.guild)}${if (u.idLong == FLOID) "\nJa Flo, du sollst dich auch bei ihm melden du Kek! :^)" else ""}
    """.trimIndent()
            ).queue()
        }
    }
}